commcare-support-library/src/main/java/org/commcare/commcaresupportlibrary/BiometricUtils.java

```diff
@@ -0,0 +1,68 @@
+package org.commcare.commcaresupportlibrary;
+
+import android.util.Base64;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BiometricUtils {
+
+    public enum BiometricIdentifier {
+        RIGHT_THUMB, RIGHT_INDEX_FINGER, RIGHT_MIDDLE_FINGER, RIGHT_RING_FINGER, RIGHT_PINKY_FINGER,
+        LEFT_THUMB, LEFT_INDEX_FINGER, LEFT_MIDDLE_FINGER, LEFT_RING_FINGER, LEFT_PINKY_FINGER,
+        FACE, UNKNOWN, INVALID
+    }
+
+    /**
+     * This method converts the biometrics templates into a Base64 encoded String to send back to
+     * CommCare, as part of the response of the callout
+     * @param templates Map containing biometric templates
+     * @return Base64 encoded string
+     */
+    public static String convertMapTemplatesToBase64String(Map<BiometricUtils.BiometricIdentifier,
+            byte[]> templates){
+        // In order to reduce the size of the templates, we are converting each byte array into a
+        // Base64 encoded String
+        Map<Integer, String> templatesBase64Encoded = new HashMap<>(templates.size());
+        for (Map.Entry<BiometricUtils.BiometricIdentifier, byte[]> template : templates.entrySet()) {
+            templatesBase64Encoded.put(template.getKey().ordinal(),
```

Comment:
any reason we are double encoding templates, here and then later again at the end of this method ?

Reply:
This is because `JSON` is used to represent the Map object, and considering that the values are byte arrays, the first encoding avoids representing them in JSON, which would increment the size due to the comas used to split the elements.

Reply:
I am struggling to understand how will this gets used in the form. With current design, Seems like we are going to store into form the whole `template` string instead of storing the individual templates in individula field like Simprint. Any reason we are deviating away from Simprints pattern ?

Reply:
@shubham1g5 the principle here is to use CC just to store the biometric templates and in the most simple way, so in a case property. any biometric operations will be preceded by the retrieval and restoration of the templates to their original format.

Reply:
I meant why not store individiual templates in separate fields in form like [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/provider/SimprintsCalloutProcessing.java#L95) ?

Reply:
It's more to avoid having to handle different form questions and case properties, when there is no apparent advantage to doing that. From my understanding, we are not doing anything with the biometric templates in CC. But I think that for the sake of standartization we could adopt the same pattern, but I wonder if you see any other gains from this.

Reply:
If we store it together and an app is capturing multiple templates, how will they separate this data out in future without writing a script ? I think storing them seprately makes any future data management on those templates much simpler for project teams.